From upstream CVS. This was fixed in 2.6.1.

Log entry:

  "Items read from a per-directory .cvsignore file should be added to the
   local_exclude_list, not the exclude_list."

Index: flist.c
===================================================================
RCS file: /cvsroot/rsync/flist.c,v
retrieving revision 1.164
retrieving revision 1.165
diff -u -r1.164 -r1.165
--- flist.c	13 Jan 2004 06:27:30 -0000	1.164
+++ flist.c	17 Jan 2004 01:16:49 -0000	1.165
@@ -942,7 +942,7 @@
 	if (cvs_exclude) {
 		if (strlen(fname) + strlen(".cvsignore") <= MAXPATHLEN - 1) {
 			strcpy(p, ".cvsignore");
-			add_exclude_file(&exclude_list,fname,MISSING_OK,ADD_EXCLUDE);
+			add_exclude_file(&local_exclude_list,fname,MISSING_OK,ADD_EXCLUDE);
 		} else {
 			io_error |= IOERR_GENERAL;
 			rprintf(FINFO,
